opa/opa.go

```diff
@@ -75,14 +81,14 @@ func (a *restAuthorizer) Authorize(subject string, groups []string, permission r
 	if err != nil {
 		level.Error(a.logger).Log("msg", "failed to marshal OPA input to JSON", "err", err.Error())
 
-		return http.StatusForbidden, false
+		return http.StatusForbidden, false, ""
 	}
 
 	res, err := a.client.Post(a.url.String(), "application/json", bytes.NewBuffer(j))
 	if err != nil {
 		level.Error(a.logger).Log("msg", "make request to OPA endpoint", "URL", a.url.String(), "err", err.Error())
 
-		return res.StatusCode, false
+		return res.StatusCode, false, ""
```

Comment:
There are a couple of cases where a response can be nil, e.g. missing `Location` header. Drilling down into the `client.do` implementation, the usual suspect on error to check for a non nil response is if `client.send` fails hard to send anything to the other side.

Reply:
So should we check if `res` is nil rather than panic? Since I hit it couple of times during local tests, I guess it would make sense to make this part bullet-proof.

Reply:
I have already updated the PR to do this.

Reply:
:face_with_head_bandage: Shouldn't leave reviews for the end of the day... Thanks, resolving!